src/algorithm/dynamics.hpp

```diff
@@ -103,10 +103,48 @@ namespace pinocchio
     a.noalias() = J.transpose() * lambda_c;
     cholesky::solve(model, data, a);
     a += data.torque_residual;
+
     
     return a;
   }
-  
+
+  ///
+  /// \brief Computes the inverse of the KKT matrix for dynamics with contact constraints, [[M JT], [J 0]].
+  /// The matrix is defined when we call forwardDynamics/impulsedynamics. This method makes use of
+  /// the matrix decompositions performed during the forwardDynamics/impulasedynamics and returns the inverse.
+  /// The jacobian should be the same that was provided to forwardDynamics/impulasedynamics.
+  /// Thus you should call forward Dynamics/impulsedynamics first.
+  /// \param[in] model The model structure of the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[out] MJtJ_inv inverse of the MJtJ matrix.
+  ///
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl,
+           typename ConstraintMatrixType>
+  inline void getKKTContactDynamicMatrixInverse(const ModelTpl<Scalar,Options,JointCollectionTpl> &
+                             model,
+                             const DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                             const Eigen::MatrixBase<ConstraintMatrixType> & J,
+                             typename DataTpl<Scalar,Options,JointCollectionTpl>::MatrixXs& MJtJ_inv)
```

Comment:
This is my final remark, I think to me as generic as possible, the resulting matrix (which stores the result) should be something like EIgen::MatrixBase<ReturnMatrixType>

Reply:
Done.